.travis.yml

```diff
@@ -1,47 +1,92 @@
- - sudo apt-add-repository -y ppa:asolovets/backports
- - sudo apt-get update -qq
- - sudo apt-get install -qq openssl libssl-dev autoconf automake protobuf-compiler libprotobuf-java libprotobuf-dev python-dev libjson-c-dev libgoogle-glog-dev libgflags-dev libldns-dev libstdc++-4.8-dev libleveldb-dev libsnappy-dev libgoogle-perftools-dev
-# Stupid frikkin' google-mock package on Precise is b0rked, so hack it up:
- - wget https://googlemock.googlecode.com/files/gmock-1.7.0.zip -O /tmp/gmock-1.7.0.zip
- - unzip -d /tmp /tmp/gmock-1.7.0.zip
- - sudo mv /tmp/gmock-1.7.0 /usr/src/gmock
- - wget https://github.com/ellzey/libevhtp/archive/1.2.10.zip -O /tmp/libevhtp-1.2.10.zip
- - unzip -d /tmp/ /tmp/libevhtp-1.2.10.zip
- - mkdir /tmp/libevhtp-1.2.10/build
- - pushd /tmp/libevhtp-1.2.10/build && cmake -DEVHTP_DISABLE_REGEX:STRING=ON -DCMAKE_C_FLAGS="-fPIC" .. && sudo make install; popd
- - sudo pip install -r python/requirements.txt
- - sudo ln -s protobuf-java.jar /usr/share/java/protobuf.jar
+# Set compiler version overrides if we have any:
+ - if [ "${REAL_CC}" != "" ]; then export CC=${REAL_CC} CXX=${REAL_CXX}; fi
+ - if [ "$TRAVIS_OS_NAME" == "osx" ]; then . ./build/travis.osx.before_install.sh; fi
+ - mkdir -p ${INSTALL_DIR}/include ${INSTALL_DIR}/lib ${INSTALL_DIR}/java
+ - echo "INSTALL_DIR=${INSTALL_DIR}"
  - mkdir -p $GOPATH/src/github.com/google
- - ln -s $PWD $GOPATH/src/github.com/google
+ - ln -sf $PWD $GOPATH/src/github.com/google
+ - echo "jars.protobuf = ${INSTALL_DIR}/java/protobuf.jar" > local.properties
+ - echo "protoc.bin = ${PROTOC}" >> local.properties
+
+install:
+ - autoconf --version
+ - pushd ..
+ - mkdir -p ct
+ - wget http://central.maven.org/maven2/com/google/protobuf/protobuf-java/2.6.0/protobuf-java-2.6.0.jar -O ${INSTALL_DIR}/java/protobuf.jar
+ - if [ ! -e depot_tools/gclient ]; then git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git; fi
+ - export PATH="$PATH:/usr/local/bin:"`pwd`/depot_tools
+ - cd ct
+ - ln -s ../certificate-transparency certificate-transparency
+ - if [ ! -f .gclient ]; then gclient config --name="certificate-transparency" --unmanaged https://github.com/${TRAVIS_REPO_SLUG}.git; fi
+ - gclient sync
+ - popd
+ - export PYTHONPATH=$(python -m site --user-site)
+ - pip install --user --upgrade -r python/requirements.txt
 # Sigh, go get -t will fetch part of this package, but doesn't catch the transitive deps so grab it manually:
  - go get github.com/stretchr/testify
  - go get -v -d -t ./...
- - java -version
- - javac -version
- - go version
- - openssl version
-
-install:
- - ./autogen.sh
 
 script:
- - ./configure && make -j$(getconf _NPROCESSORS_ONLN) check VERBOSE=1
+ - make -j$(getconf _NPROCESSORS_ONLN) check VERBOSE=1
  - ant build test
  - make -C python test
- - go test -v ./go/...
+ - pushd ${GOPATH}
+ - LD_LIBRARY_PATH=${INSTALL_DIR}/lib CGO_CPPFLAGS="-I${INSTALL_DIR}/include" CGO_LDFLAGS="-L${INSTALL_DIR}/lib" go test -v ./src/...
```

Comment:
An issue for another time: we need to make the content of `${INSTALL_DIR}/lib` only be static libraries, and skip this `LD_LIBRARY_PATH`nonsense. This way lies madness...